std_traffic/utils/data_utils.py

```diff
@@ -44,29 +44,72 @@ def get_db(
         cursor_factory=psycopg2.extras.DictCursor)
 
 
-def get_columns(csv_file: str, delim: str = ';', nrows: int = 100) -> dict:
-    """Extract the column names from a CSV file.
+def get_columns(csv_file: str, delim: str = ';') -> dict:
+    """Extract the column names and types from a CSV file.
     The default choice for the delimiter (;) is related to SUMO defaults.
-    By default use the first 100 columns to understand the type.
 
     :param csv_file: (str) Path to the CSV.
     :param delim: (str) Delimiter character used in the CSV (default ;).
-    :param nrows: (int) Number of rows to use to infer the types (default 100).
     :return: (dict) Map column_name -> column_type. Types are Numpy's.
     """
 
     # use pandas features
-    dataf = pd.read_csv(csv_file, nrows=nrows, sep=delim)
-    all_types = dict(dataf.dtypes)
-    return all_types
+    # - Read first 1000 row, get columns names, infer types
+    #   by avoiding NA columns
+    # - For each column type missing
+    #   keep reading only that column in chunks until a type is found
+    #   or EOF
+    initial = 1000
+    dataf = pd.read_csv(csv_file, sep=delim, nrows=initial)
+    all_names = set(dataf.columns)
```

Comment:
@RuggeroFabbiano Search operation (like `in` in Python) is O(1) with sets and O(n) with lists, like `dataf.columns` would be.
However, looking at the rest of the function, this cast is basically useless because the search in the set is done in one place only. And anyway the size of list of always relatively small (it's the number of features, after all).
It was more of an instinctive reflex to convert it to `set` because I knew I was going to search in it.
Do you think we should remove it?

Reply:
Since the search is done only once I'd say "yes", given that the conversion is also _O_(_n_).

Reply:
Fixed in 1756eae